challenge_demo/test/grammar_test.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import print_function
```

Comment:
TBH: the logic of the code looks good to me, however I do not completely understand the working